psl-core/src/main/java/org/linqs/psl/util/VizDataCollection.java

```diff
@@ -0,0 +1,233 @@
+            e.printStackTrace();
+        }
+    }
+
+    private static class ShutdownHook extends Thread {
+        @Override
+        public void run() {
+            outputJSON();
+        }
+    }
+
+    //Takes in a prediction truth pair and adds it to our map
+    public static void predictionTruth(GroundAtom target, float predictVal, float truthVal ) {
+        JSONObject moduleElement = new JSONObject();
+        moduleElement.put("Truth", truthVal);
+        moduleElement.put("Prediction", predictVal);
+        moduleElement.put("Predicate", target.toString());
+        vizData.predictionTruthArray.put(moduleElement);
+    }
+
+     public static void groundingsPerRule(List<Rule> rules, GroundRuleStore groundRuleStore) {
+
+        for (Rule rule: rules)
+        {
+            String stringRuleId = Integer.toString(System.identityHashCode(rule));
+            int groundRuleCount = groundRuleStore.count( rule );
+            if ( !vizData.rules.isNull(stringRuleId) ) {
+                JSONObject ruleElement = vizData.rules.getJSONObject(stringRuleId);
+                ruleElement.put("count", groundRuleCount);
+                ruleElement.put("weighted", rule.isWeighted());
+            }
+            else {
+                JSONObject newRuleElement = new JSONObject();
+                newRuleElement.put("string", rule.getName());
+                newRuleElement.put("count", groundRuleCount);
+                newRuleElement.put("weighted", rule.isWeighted());
+                vizData.rules.put(stringRuleId, newRuleElement);
+            }
+        }
+     }
+
+    public static void totalRuleSatDis(List<Rule> rules, GroundRuleStore groundRuleStore) {
+        for (Rule rule : rules) {
+            Iterable<GroundRule> groundedRuleList = groundRuleStore.getGroundRules(rule);
+            double totalSat = 0.0;
+            double totalDis = 0.0;
+            int groundRuleCount = 0;
+            JSONObject moduleElement = new JSONObject();
+
+            for (GroundRule groundRule : groundedRuleList) {
+                if (groundRule instanceof WeightedGroundRule) {
+                    WeightedGroundRule weightedGroundRule = (WeightedGroundRule)groundRule;
+                    totalSat += 1.0 - weightedGroundRule.getIncompatibility();
+                    totalDis += weightedGroundRule.getIncompatibility();
+                }
+                groundRuleCount++;
+            }
+            moduleElement.put("Rule", rule.getName());
+            moduleElement.put("Total Satisfaction", totalSat);
+            moduleElement.put("Satisfaction Percentage", totalSat / groundRuleCount);
+            moduleElement.put("Total Dissatisfaction", totalDis);
+            moduleElement.put("Dissatisfaction Percentage", totalDis / groundRuleCount);
+            vizData.totRuleSatArray.put(moduleElement);
+        }
+    }
+
+    public static void violatedGroundRules(List<Rule> rules, GroundRuleStore groundRuleStore) {
```

Comment:
Move to UI.

Reply:
So remove the violatedGroundRules and totalRuleSatDis methods because we can compute the data in the visualization?

Reply:
Yup.